erpnext/manufacturing/report/discrete_manufacture_costing/discrete_manufacture_costing.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2013, halima and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from erpnext import get_company_currency, get_default_company
+
+def execute(filters=None):
+    columns = get_column()
+    data, data_to_be_printed  = get_data(filters)
+    return columns, data, [], [], data_to_be_printed 
+
+def get_column():
+    return [
+        _("Sales Order") + ":Link/Sales Order:100",
+        _("Work Order") + ":Link/Work Order:100",
+        _("Production Plan") + ":Link/Production Plan:100",
+        _("Item Name") + ":Data/Item Name:300",
+        _("Stock Entry Type") + ":Data/Stock Entry Type:100",
+        _("Stock Item Group") + ":Data/Stock Item Group:100",
+        _("Quantity") + ":Data/Quantity:100",
+        _("UOM") + ":Data/UOM:100",
+        _("Valuation Rate") + ":Data/Valuation Rate:100",
+        _("Amount") + ":Data/Amount:100",
+    ]
+
+def get_data(filters):
+    sale_order = filters.salesOrder
+    production_plan = filters.productionPlan
+    item_group = filters.itemGroup
+    filterItemGroup = tuple(item_group)
+    stock_entry_type = "Manufacture"
+    # fetch report columns value
+    if(filterItemGroup):
+        response = frappe.db.sql(""" SELECT SOD.sales_order, WO.name, WO.production_plan, 
+                SED.item_code, SE.stock_entry_type, SED.item_group, 
+                SUM(ROUND(SED.qty, 3)), SED.uom, SED.basic_rate,
+                (SED.qty * SED.basic_rate) as amount
+            FROM `tabWork Order` as WO
+                JOIN `tabStock Entry` as SE
+                    ON SE.work_order = WO.name
+                JOIN `tabStock Entry Detail` as SED
+                    ON SED.parent = SE.name
+                JOIN `tabProduction Plan` as PP
```

Comment:
Many users does not use "Production Plan" in discrete manufacturing. It should not be involved in this report. We can directly get all Work Order against Sales Order based on "sales_order" field inside work order.